### PR TITLE
[improvement] Add option to not create run spans in AsyncTracer

### PR DESCRIPTION
**Pull request by robert3005**
## Before this PR
AsyncTracer would always create two spans "-enqeueu" and "-run"

## After this PR
Users can indicate whether they want the additional "-run" span

### CODE REVIEW

File: tracing/src/main/java/com/palantir/tracing/AsyncTracer.java [hunk 1]
```diff
@@ -56,12 +57,21 @@ public AsyncTracer(String operation) {
         this(Optional.of(operation));
     }
 
+    public AsyncTracer(String operation, boolean createRunSpan) {
+        this(Optional.of(operation), createRunSpan);
```
Comment (carterkozak) [inline tracing/src/main/java/com/palantir/tracing/AsyncTracer.java:61]: this ctor should house the full implementation, optional ctor just passes `operation.orElse(DEFAULT_OPERATION)` for backcompat

**Review by carterkozak — COMMENTED** [review]
This is more confusing than #124

File: tracing/src/main/java/com/palantir/tracing/AsyncTracer.java [hunk 2]
```diff
@@ -76,12 +86,17 @@ public AsyncTracer(Optional<String> operation) {
         Tracer.setTrace(deferredTrace);
         // Finish the enqueue span
         Tracer.fastCompleteSpan();
-        Tracer.startSpan(operation + "-run");
```
Comment (carterkozak) [inline tracing/src/main/java/com/palantir/tracing/AsyncTracer.java:79]: bug: `-run` doesn't always have the same traceId as `-enqueue`

**Comment by robert3005** [comment]
You're right - I forgot the conversation we had about this previously

**Pull request closed without merging**